### ISSUE TITLE
--specpath not working

### ISSUE THREAD

**Issue by Xoriun** [pyinstaller/pyinstaller#8356]
When running pyinstaller from console, the ```--specpath``` option is ignored.
When pyinstaller has no write-permissions in the root dir, it crashes when trying to create the ```.spec``` file:

```
772 INFO: PyInstaller: 6.5.0, contrib hooks: 2024.3
773 INFO: Python: 3.11.6
802 INFO: Platform: Windows-10-10.0.19045-SP0
804 INFO: wrote C:/Users/-/Documents/systec/Flowcom/tests/p2e.spec
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\-\AppData\Local\Programs\Python\Python311\Scripts\pyinstaller.exe\__main__.py", line 7, in <module>
  File "C:\Users\-\AppData\Local\Programs\Python\Python311\Lib\site-packages\PyInstaller\__main__.py", line 228, in _console_script_run
    run()
  File "C:\Users\-\AppData\Local\Programs\Python\Python311\Lib\site-packages\PyInstaller\__main__.py", line 212, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "C:\Users\-\AppData\Local\Programs\Python\Python311\Lib\site-packages\PyInstaller\__main__.py", line 69, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "C:\Users\-\AppData\Local\Programs\Python\Python311\Lib\site-packages\PyInstaller\building\build_main.py", line 1139, in main
    build(specfile, distpath, workpath, clean_build)
  File "C:\Users\-\AppData\Local\Programs\Python\Python311\Lib\site-packages\PyInstaller\building\build_main.py", line 1040, in build
    os.makedirs(pth, exist_ok=True)
  File "<frozen os>", line 225, in makedirs
PermissionError: [WinError 5] Access is denied: 'C:\\windows\\system32\\dist'
```

Running dev build of pyinstaller (6.5.0) on windows 10, python version is 3.11.6.

**Comment by rokm** [comment]
What was the PyInstaller command in the above example?

**Comment by bwoodsend** [comment]
And are you really running PyInstaller from inside `C:\\windows\\system32`?

**Comment by rokm** [comment]
> 804 INFO: wrote C:/Users/-/Documents/systec/Flowcom/tests/p2e.spec

Seems to me spec was created where you asked via `--specpath`.

> PermissionError: [WinError 5] Access is denied: 'C:\\windows\\system32\\dist'

But the `build` directory and the `dist` directory are created in current working directory, unless you also override them with [`--workpath`](https://pyinstaller.org/en/stable/usage.html#cmdoption-workpath) and [`--dispath`](https://pyinstaller.org/en/stable/usage.html#cmdoption-distpath), respectively.

**Comment by Xoriun** [comment]
Ok, I have no idea what I did one hour ago...
I tried to use ```--specpath``` and it didn't work, so I created a small problem and tried to run pyinstaller from console (I just hit win-r cmd, that's why pyinstaller was running from system32) and then got that error.
I closed the cmd window and didn't save the command (stupid me) and when I tried to recreate the command just now, it just worked. I have no idea whatsoever what I did differently.
If I manage to get the same error again, I will do another post here, but for now I'm going to close this.
